pybamm/pybamm_data.py

```diff
@@ -0,0 +1,137 @@
+import pooch
+import pathlib
+
+
+class DataLoader:
+    """
+    Data Loader class for downloading and loading data files upstream at https://github.com/pybamm-team/pybamm-data/
+
+    Attributes
+    ----------
+    path: POSIX/Windows PATH
+       POSIX/Windows path to the data file.
+       The folder locations are defined by the `platformdirs <https://platformdirs.readthedocs.io/en/latest/>`_ package using the `user_cache_dir <https://platformdirs.readthedocs.io/en/latest/api.html#cache-directory>`_ function. The default cache paths for pybamm are located under -
+
+        - Mac: ``~/Library/Caches/v1.x.x/pybamm``
+
+        - Unix: ``~/.cache/v1.x.x/pybamm`` or the value of the XDG_CACHE_HOME environment variable, if defined.
+
+        - Windows: ``C:\\Users\\<user>\\AppData\\Local\\pybamm\\v1.x.x\\Cache``
+    registry: pooch.core.Pooch object
+        Creates a registry of data files available to fetch using pooch.
+    """
+
+    def __init__(self):
+        """
+        Create a pooch registry with the following data files available upstream at https://github.com/pybamm-team/pybamm-data/
+
+        COMSOL RESULTS
+        ---------------
+        comsol_01C.json
+        comsol_05C.json
+        comsol_1C.json
+        comsol_1plus1D_3C.json
+        comsol_2C.json
+        comsol_3C.json
+
+        Kokam SLPB 75106100 discharge data from Ecker et al (2015)
+        ----------------------------------------------------------
+        Ecker_1C.csv
+        Ecker_5C.csv
```

Comment:
Could you add the relevant papers here (and similarly for other data files under the other headings where applicable)? They should be present in the `CITATIONS.bib` file. If they are not, you can get the BibTeX reference for them from Google Scholar or similar and add them to the file. Then, you can reference them in the docstring with something like

```rest

:footcite:t:`AuthorYear`

```

Reply:
Sure!